pkg/app/server/grpcapi/web_api.go

```diff
@@ -1530,68 +1526,55 @@ func (a *WebAPI) ListAPIKeys(ctx context.Context, req *webservice.ListAPIKeysReq
 
 // GetInsightData returns the accumulated insight data.
 func (a *WebAPI) GetInsightData(ctx context.Context, req *webservice.GetInsightDataRequest) (*webservice.GetInsightDataResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "Not available")
-
-	// claims, err := rpcauth.ExtractClaims(ctx)
-	// if err != nil {
-	// 	a.logger.Error("failed to authenticate the current user", zap.Error(err))
-	// 	return nil, err
-	// }
-
-	// if math.Abs(float64(req.Offset)) > 86400 {
-	// 	return nil, status.Error(codes.InvalidArgument, "offset must between -86400 ~ 86400")
-	// }
-
-	// loc := time.FixedZone("", int(req.Offset))
-
-	// switch req.MetricsKind {
-	// case model.InsightMetricsKind_DEPLOYMENT_FREQUENCY:
-	// 	points, updatedAt, err := a.getDeploymentFrequency(ctx, claims.Role.ProjectId, req.ApplicationId, req.RangeFrom, req.RangeTo, loc)
-	// 	if err != nil {
-	// 		return nil, err
-	// 	}
-	// 	return &webservice.GetInsightDataResponse{
-	// 		UpdatedAt: updatedAt,
-	// 		Type:      model.InsightResultType_MATRIX,
-	// 		Matrix: []*model.InsightSampleStream{
-	// 			{DataPoints: points},
-	// 		},
-	// 	}, nil
-	// default:
-	// 	return nil, status.Error(codes.Unimplemented, "")
-	// }
-}
-
-// getDeploymentFrequency if applicationID is empty, collect all application
-func (a *WebAPI) getDeploymentFrequency(ctx context.Context, projectID string, applicationID string, from, to int64, loc *time.Location) ([]*model.InsightDataPoint, int64, error) {
-	const deploymentFrequencyMinimumVersion = model.InsightDeploymentVersion_V0
-
-	deployments, err := a.insightStore.List(ctx, projectID, from, to, deploymentFrequencyMinimumVersion)
-	var updatedAt int64
-	if err != nil {
-		return nil, updatedAt, err
-	}
-
-	if applicationID != "" {
-		filtered := deployments[:0]
-		for _, v := range deployments {
-			if v.AppId == applicationID {
-				filtered = append(filtered, v)
-			}
-		}
-		deployments = filtered
+	claims, err := rpcauth.ExtractClaims(ctx)
+	if err != nil {
+		a.logger.Error("failed to authenticate the current user", zap.Error(err))
+		return nil, err
+	}
+
+	if req.MetricsKind != model.InsightMetricsKind_DEPLOYMENT_FREQUENCY &&
```

Comment:
This process looks to be duplicated with [this](https://github.com/pipe-cd/pipecd/pull/3993/files#diff-ff91a92f1a5375d64c7dd4a461b01f9712f51e2aa5a5ba5abe2327d969874e5fR1564), right?

Reply:
Right. Let me delete this part.